src/aind_data_asset_indexer/models.py

```diff
@@ -109,3 +109,64 @@ class AindIndexBucketsJobSettings(AindIndexBucketJobSettings):
     # Set individual bucket off
     s3_bucket: type(None) = None
     s3_buckets: List[str]
+
+
+class CodeOceanIndexBucketJobSettings(IndexJobSettings):
+    """Aind Index Bucket Job Settings"""
+
+    doc_db_host: str
+    doc_db_port: int
+    doc_db_user_name: str
+    doc_db_password: SecretStr
+    doc_db_db_name: str
+    doc_db_collection_name: str
+    codeocean_domain: str
+    codeocean_token: SecretStr
+
+    @classmethod
+    def from_param_store(cls, param_store_name: str):
+        """
+        Construct class from aws param store and secrets manager
+        Parameters
+        ----------
+        param_store_name : str
+        """
+        param_store_client = boto3.client("ssm")
+        response = param_store_client.get_parameter(
+            Name=param_store_name, WithDecryption=True
+        )
+        param_store_client.close()
+        parameters = response["Parameter"]["Value"]
+        parameters_json = json.loads(parameters)
+        if "doc_db_secret_name" not in parameters:
```

Comment:
nit: I was checking values from the `parameters` string and not the `parameters_json` keys in `AindIndexBucketJobSettings`. It works as-is but is not ideal. Let me know if you want me to fix this in both models.

```suggestion
        if "doc_db_secret_name" not in parameters_json:
```

Reply:
I'll fix it. I think you needed to check `parameters_json.keys()`

Reply:
Thanks!